java/src/jmri/util/usb/UsbBrowserPanel.java

```diff
@@ -134,18 +141,58 @@ private void buildTree(UsbTreeNode root, int depth) {
                 root.add(node);
             }
         }
-        // prevent NPE if called in constructor
+        // prevent NPE when called in constructor
         if (usbTree != null) {
             TreePath selection = usbTree.getSelectionPath();
             ((DefaultTreeModel) usbTree.getModel()).reload(root);
             usbTree.setSelectionPath(selection);
         }
     }
 
+    public void expandAll(@Nullable JTree tree) {
+        expandAll(tree, true);
+    }
+
+    public void unexpandAll(@Nullable JTree tree) {
+        expandAll(tree, false);
+    }
+
+    public void expandAll(@Nullable JTree tree, boolean expand) {
+        if (tree != null) {
+            TreeNode root = (TreeNode) tree.getModel().getRoot();
+            expandAll(tree, new TreePath(root), expand);
+
+            // if we just collapsed everything...
+            if (!expand) {
+                // reexpand the root TreeNode
+                tree.expandPath(new TreePath(root));
+            }
+        }
+    }
+
+    private static void expandAll(JTree tree, TreePath parentTreePath, boolean expand) {
```

Comment:
Why is this private static method not just code within the public non-static method two argument expandAll method?

Reply:
Because it has to recourse (with three parameters).